planb/create_cluster.py

```diff
@@ -686,13 +694,121 @@ def create_cluster(options: dict):
         # Undo stack sounds like a natural choice.
         #
         for region, sg in security_groups.items():
-            ec2 = boto3.client('ec2', region)
+            ec2 = ec2_client(region)
             info('Cleaning up security group: {}'.format(sg['GroupId']))
             ec2.delete_security_group(GroupId=sg['GroupId'])
 
         if options['use_dmz']:
             for region, ips in node_ips.items():
-                ec2 = boto3.client('ec2', region)
+                ec2 = ec2_client(region)
+                for ip in ips:
+                    info('Releasing IP address: {}'.format(ip['PublicIp']))
+                    ec2.release_address(AllocationId=ip['AllocationId'])
+
+        raise
+
+
+def extend_cluster(options: dict):
+    options['regions'] = [options['region']]
+
+    ec2 = ec2_client(options['region'])
+    running_instances = [
+        i
+        for i in list_instances(ec2, options['cluster_name'])
+        if i['State']['Name'] == 'running'
```

Comment:
Couldn't this be an option for `list_instances` to filter for a state instead doing it in the command?